core/evm.go

```diff
@@ -61,6 +61,8 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 		Difficulty:  new(big.Int).Set(header.Difficulty),
 		BaseFee:     baseFee,
 		GasLimit:    header.GasLimit,
+		Random:      header.MixDigest,
```

Comment:
Nitpick, I think we should keep this as `nil` before the merge, just to catch any code accidentally touching this field prematurely.